From e624cd1959904735eda81f1a9a45d99848bb3503 Mon Sep 17 00:00:00 2001
From: Chunwei Chen <tuxoko@gmail.com>
Date: Tue, 9 May 2017 10:38:46 -0700
Subject: [PATCH] Linux 4.12 compat: PF_FSTRANS was removed
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

zfsonlinux/spl@8f87971 added __spl_pf_fstrans_check for the xfs related
check, so we use them accordingly.

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Chunwei Chen <david.chen@osnexus.com>
Closes #6113
(backport of upstream zfs commit e624cd1959904735eda81f1a9a45d99848bb3503)
Signed-off-by: Colin Ian King <colin.king@canonical.com>

Index: zfs-linux-0.6.5.9/include/sys/zfs_context.h
===================================================================
--- zfs-linux-0.6.5.9.orig/include/sys/zfs_context.h
+++ zfs-linux-0.6.5.9/include/sys/zfs_context.h
@@ -748,6 +748,7 @@ typedef int fstrans_cookie_t;
 extern fstrans_cookie_t spl_fstrans_mark(void);
 extern void spl_fstrans_unmark(fstrans_cookie_t);
 extern int spl_fstrans_check(void);
+extern int __spl_pf_fstrans_check(void);
 
 #endif /* _KERNEL */
 #endif	/* _SYS_ZFS_CONTEXT_H */
Index: zfs-linux-0.6.5.9/lib/libzpool/kernel.c
===================================================================
--- zfs-linux-0.6.5.9.orig/lib/libzpool/kernel.c
+++ zfs-linux-0.6.5.9/lib/libzpool/kernel.c
@@ -1319,7 +1319,7 @@ spl_fstrans_unmark(fstrans_cookie_t cook
 }
 
 int
-spl_fstrans_check(void)
+__spl_pf_fstrans_check(void)
 {
 	return (0);
 }
Index: zfs-linux-0.6.5.9/module/zfs/vdev_file.c
===================================================================
--- zfs-linux-0.6.5.9.orig/module/zfs/vdev_file.c
+++ zfs-linux-0.6.5.9/module/zfs/vdev_file.c
@@ -200,7 +200,7 @@ vdev_file_io_start(zio_t *zio)
 			 * already set, see xfs_vm_writepage().  Therefore
 			 * the sync must be dispatched to a different context.
 			 */
-			if (spl_fstrans_check()) {
+			if (__spl_pf_fstrans_check()) {
 				VERIFY3U(taskq_dispatch(system_taskq,
 				    vdev_file_io_fsync, zio, TQ_SLEEP), !=, 0);
 				return;
